chat-plugins/datasearch.js

```diff
@@ -675,8 +675,9 @@ function runDexsearch(target, cmd, canAll, message) {
 			}
 
 			if (alts.tiers && Object.keys(alts.tiers).length) {
-				if (alts.tiers[dex[mon].tier]) continue;
-				if (Object.values(alts.tiers).includes(false) && alts.tiers[dex[mon].tier] !== false) continue;
+				let tier = alts.tiers[dex[mon].tier] || alts.tiers[dex[mon].tier.slice(1, -1)];
+				if (tier) continue;
+				if (Object.values(alts.tiers).includes(false) && tier !== false) continue;
```

Comment:
Sorry for the delay, I blame the GitHub UI for not making a @mention look more important than a regular comment.

So, the way these search categories work is as follows: `alts.tiers` is an object whose keys are the tiers in the query and whose values are `true` for a regular query or `false` for a `!` query. For example, if I write `/ds Uber | OU` then `alts.tiers = {OU: true, UU:true}` while if I write `/ds !Uber` then `alts.tiers = {Uber: false}`.

So the first step is easy: just check wither `alts.tiers[dex[mon].tier]` is `true`. If so, then this is a `Uber | OU` type query, and we've found a matching tier, so we can continue with the next test.

If however it's not `true`, then there are three possibilities: i) an `Uber | OU` type query after all, so the match has failed ii) an `!Uber` type query with `alts.tiers[dex[mon].tier]` is `false`, which again is a failure iii) an `!Uber` type qurey with `alts.tiers[dex[mon].tier]` is `undefined`, which is a success (since that means that that tier was not excluded).

(It occurs to me that the above might simplify to `if (alts.tiers[dex[mon].tier] === Object.values(alts.tiers)[0]) continue;`, but if that's a good idea here then it should probably be done for gen and colour too.)

Now your problem is that you're checking `alts.tiers[dex[mon].tier] || alts.tiers[dex[mon].tier.slice(1, -1)]`, so if it's a `!Uber` type query and the left value is actually `false` then it will get ignored and the code will think that the match succeeded when in fact it failed (because the tier is one that should have been excluded).